10_rest/willshen8/pkg/puppy/mapstore.go

```diff
@@ -0,0 +1,76 @@
+package puppy
+
+import (
+	"strconv"
+)
+
+type MapStore struct {
+	ms     map[uint32]Puppy
+	nextID uint32
+}
+
+// NewMapStore initialise a new MapStore
+func NewMapStore() *MapStore {
+	return &MapStore{ms: map[uint32]Puppy{}}
+}
+
+// CreatePuppy create a new puppy and store in mapStore.
+func (m *MapStore) CreatePuppy(p *Puppy) (uint32, error) {
+	if i, err := strconv.Atoi(p.Value); err == nil {
+		if i < 0 {
+			return 0, &Error{
+				Message: PuppyValueLessThanZero,
+				Code:    NegativeValue,
+			}
+		}
+	} else {
+		return 0, &Error{
+			Message: InvalidPuppyValue,
+			Code:    ErrorValueFormat,
+		}
+	}
```

Comment:
The logic here is that I will read the value of the puppy, if it's not even a numerical value, then I return an error. If it is a numerical value, then I will then further check if it is less than 0, if so return a different error. If it passes these two checks, then it will create a puppy.

Reply:
I know what it does but it could read nicer, pls update:
```go
      i, err := strconv.Atoi(p.Value)
      if err != nil {
        msg := fmt.Sprintf(`cannot parse puppy value "%s" as integer`, p.Value)
      	return 0, &Error{ErrBadInput, msg}
      }
      if i < 0 {
        msg := fmt.Sprintf(`puppy value "%d" cannot be negative`, i)
      	return 0, &Error{ErrBadInput, msg}
      }
```

Reply:
still not done.